rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -5275,4 +5279,373 @@ public Wildcard withBound(@Nullable JLeftPadded<Bound> bound) {
             }
         }
     }
+
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    @AllArgsConstructor
+    final class Module implements J {
+
+        @Nullable
+        @NonFinal
+        transient WeakReference<Padding> padding;
+
+        @With
+        @EqualsAndHashCode.Include
+        @Getter
+        UUID id;
+
+        @With
+        @Getter
+        Space prefix;
+
+        @With
+        @Getter
+        Markers markers;
+
+        @With
+        @Getter
+        List<Annotation> annotations;
+
+        @With
+        @Getter
+        ModuleKind moduleType;
+
+        @With
+        @Getter
+        J.FieldAccess name;
+
+        List<JRightPadded<Directive>> directives;
+
+        @Getter
+        @With
+        Space end;
+
+        public Module withDirectives(List<Directive> directives) {
+            return getPadding().withDirectives(JRightPadded.withElements(this.directives, directives));
+        }
+
+        public List<Directive> getDirectives() {
+            return JRightPadded.getElements(directives);
+        }
+
+        public enum ModuleKind {
+            OPEN,
+            STRONG
+        }
+
+        public Padding getPadding() {
+            Module.Padding p;
+            if (this.padding == null) {
+                p = new Padding(this);
+                this.padding = new WeakReference<>(p);
+            } else {
+                p = this.padding.get();
+                if (p == null || p.t != this) {
+                    p = new Padding(this);
+                    this.padding = new WeakReference<>(p);
+                }
+            }
+            return p;
+        }
+
+        @RequiredArgsConstructor
+        public static class Padding {
+            private final Module t;
+
+            public Module withDirectives(@Nullable List<JRightPadded<Directive>> directives) {
+                return t.directives == directives ? t : new Module(t.id, t.prefix, t.markers, t.annotations,
+                        t.moduleType, t.name, directives, t.end);
+            }
+
+
+            public List<JRightPadded<Directive>> getDirectives() {
+                return t.directives;
+            }
+        }
+
+        @Override
+        public <P> J acceptJava(JavaVisitor<P> v, P p) {
+            return v.visitModule(this, p);
+        }
+
+        @Override
+        public String toString() {
+            return withPrefix(Space.EMPTY).printTrimmed(new JavaPrinter<>());
+        }
+    }
+
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    final class Requires implements Directive {
+
+        @With
```

Comment:
If every field should have a `with` method, you can put the `@With` annotation on the class and it will apply to every field. Some goes for similar lombok annotations like `@Getter`, `@Setter`.